### PR TITLE
(WIP) #160577608 Send email verification link to user upon successful registration

### PR DESCRIPTION
**Pull request by gitaumoses4**
## What does this PR do?
It sends a verification link to users upon successful login.

## Description of tasks to be completed.
1. Send email to user upon login with verification link
2. Generate token to be used to identify a user.
3. Process the verification and verify the user.

## Relevant Pivotal Tracker Stories
[#160577608](https://www.pivotaltracker.com/story/show/160577608)

## Screenshots
With local testing, I have managed to check the UI for the email received by the user.

screen shot 2018-09-28 at 00 27 16

When a user successfully registers, the user receives a message as follows:

screen shot 2018-10-01 at 14 25 43

On clicking the "Verify now" button, the user is redirected as follows:

screen shot 2018-10-01 at 14 27 14

If the User uses an invalid link, the user gets a message as follows: (Notice that the url has been changed from 'bevah' to 'chariss'.

screen shot 2018-10-01 at 14 52 38

## Any background context
Sending HTML email requires using inline CSS which can be tedious. Therefore, I managed to find a django package that automatically embeds the inline CSS from the external css file.
The library can be found here
https://github.com/roverdotcom/django-inlinecss

### CODE REVIEW

**Comment by C3real-kill3r** [comment]
This is so good, It's encouraging work

**Review by C3real-kill3r — APPROVED** [review]

**Comment by charisschomba** [comment]
I have tested it and it is working.

File: requirements.txt [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
 PyJWT==1.6.4
+pynliner3==0.6
 pytz==2018.5
+requests==2.19.1
+simplejson==3.16.0
 six==1.11.0
-django-rest-swagger==2.2.0
+uritemplate==3.0.0
+urllib3==1.23
 whitenoise==4.1
```
Comment (charisschomba) [inline requirements.txt:14]: You only need to include the main dependency that installs  all other sub-dependencies in this case you could have only  included `django-inlinecss==0.1.2`

File: authors/apps/authentication/email_verification.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                'domain': current_site.domain,
+                'uid':urlsafe_base64_encode(force_bytes(user.pk)),
+                'token':account_activation_token.make_token(user),
+            })
+            user_email = user.email
+            email = EmailMessage(
+                        email_subject, email_message, to=[user_email]
+            )
+            email.send()
+            #i'm still looking for a way to return the token (account_activation_token) here
```
Comment (charisschomba) [inline authors/apps/authentication/email_verification.py]: You can create an activation link this way,
`activation_link = 'http://' + current_site_domain + 'api/account/activate/' + token + '/'` and send it via email
then create `/account/activate` endpoint to verify the user when he/she click the link

Comment (bevkololi) [reply authors/apps/authentication/email_verification.py]: Thanks a lot @charisschomba

File: .env_example [hunk 3]
```diff
@@ -4,4 +4,12 @@ export DB_NAME=
 export DB_USER=postgres
 export DB_PASSWORD=
 export DB_HOST=localhost
-
+export EMAIL_HOST=smtp.sendgrid.net
+export EMAIL_HOST_USER=
+export EMAIL_HOST_PASSWORD=
+<<<<<<< HEAD
```
Comment (esirK) [inline .env_example]: Kindly fix this.

File: requirements.txt (hunk 1 above)
Comment (esirK) [reply requirements.txt:14]: Have you checked out this comment??

**Comment by esirK** [comment]
Looks good and ready for merging but first work on the comments and also do some research on how to test this. You can look into how to mock the sending of emails.

File: requirements.txt (hunk 1 above)
Comment (esirK) [reply requirements.txt:14]: @gitaumoses4 ??

File: .env_example (hunk 3 above)
Comment (esirK) [reply .env_example]: @gitaumoses4

Comment (gitaumoses4) [reply .env_example]: Cool. Let me push the changes.

File: requirements.txt (hunk 1 above)
Comment (gitaumoses4) [reply requirements.txt:14]: Let me push the changes for this review. Thanks @esirK

**Comment by esirK** [comment]
Looks good. Kindly liaise with @mutaimwiti and add the env variables to Heroku.

**Pull request merged**